arangod/RocksDBEngine/RocksDBVectorIndex.cpp

```diff
@@ -379,30 +368,51 @@ Result RocksDBVectorIndex::insert(transaction::Methods& /*trx*/,
                         input.size(), _definition.dimension)};
   }
 
-  auto const docId = static_cast<faiss::idx_t>(documentId.id());
   if (_definition.metric == SimilarityMetric::kCosine) {
     faiss::fvec_renorm_L2(_definition.dimension, 1, input.data());
   }
-  try {
-    flatIndex.add_with_ids(1, input.data(), &docId);
-  } catch (basics::Exception& e) {
-    return {e.code(), e.message()};
+
+  return {};
+}
+
+/// @brief inserts a document into the index
+Result RocksDBVectorIndex::insert(transaction::Methods& /*trx*/,
+                                  RocksDBMethods* methods,
+                                  LocalDocumentId documentId,
+                                  velocypack::Slice doc,
+                                  OperationOptions const& /*options*/,
+                                  bool /*performChecks*/) {
+  std::vector<float> input;
+  if (auto res = readDocumentVectorData(doc, input); res.fail()) {
+    return res;
   }
 
+  faiss::idx_t listId{0};
+  TRI_ASSERT(_faissIndex->quantizer != nullptr);
+  _faissIndex->quantizer->assign(1, input.data(), &listId);
+
+  RocksDBKey rocksdbKey;
+  rocksdbKey.constructVectorIndexValue(objectId(), listId, documentId);
+  auto status = methods->Delete(_cf, rocksdbKey);
+
+  std::unique_ptr<uint8_t[]> flat_codes(new uint8_t[_faissIndex->code_size]);
+
+  // TODO: since we only use IVTFlat this is just copying the data.
+  //  Probably we want to use some PQ encoding later on.
```

Comment:
Is this true, since the factory can produce `IndexIVFPQ` index and that will pass the assert. And that index uses PQ encoding
```
 _faissIndex = std::unique_ptr<faiss::IndexIVF>{
        dynamic_cast<faiss::IndexIVF*>(faiss::read_index(&reader))};
    ADB_PROD_ASSERT(_faissIndex != nullptr);
```

Reply:
I think this comment is now simply outdated. We should remove it. The encoding will work correctly based on what ever index was chosen.